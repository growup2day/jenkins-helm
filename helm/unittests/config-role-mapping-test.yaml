templates:
  - config-role-mapping.yaml
tests:
  - it: by default should not generate config map
    asserts:
      - hasDocuments:
          count: 0
  - it: when enabled, content should match snapshot
    set:
      customiseRoleMapping:
        enabled: true
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: data.Overall-Administer
          value: admin
      - equal:
          path: data.Job-Build
          value: edit,admin
      - matchSnapshot: {}
  - it: should reflect the customisations made
    set:
      customiseRoleMapping:
        enabled: true
        mappings:
          Overall-Administer: edit,admin
          Job-Build: view,edit,admin
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: data.Overall-Administer
          value: edit,admin
      - equal:
          path: data.Job-Build
          value: view,edit,admin
