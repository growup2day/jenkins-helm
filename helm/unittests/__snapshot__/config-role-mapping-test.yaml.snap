when enabled, content should match snapshot:
  1: |
    apiVersion: v1
    data:
      Agent-Build: admin
      Agent-Configure: admin
      Agent-Connect: admin
      Agent-Create: admin
      Agent-Delete: admin
      Agent-Disconnect: admin
      Artifactory-Promote: admin
      Artifactory-Release: admin
      Credentials-Create: admin
      Credentials-Delete: admin
      Credentials-ManageDomains: admin
      Credentials-Update: admin
      Credentials-View: view,edit,admin
      Job-Build: edit,admin
      Job-Cancel: edit,admin
      Job-Configure: edit,admin
      Job-Create: edit,admin
      Job-Delete: edit,admin
      Job-Discover: view,edit,admin
      Job-Move: admin
      Job-Read: view,edit,admin
      Job-Workspace: edit,admin
      Metrics-HealthCheck: admin
      Metrics-ThreadDump: admin
      Metrics-View: admin
      Overall-Administer: admin
      Overall-Read: view,edit,admin
      Run-Delete: admin
      Run-Replay: admin
      Run-Update: admin
      SCM-Tag: edit,admin
      View-Configure: admin
      View-Create: admin
      View-Delete: admin
      View-Read: admin
    kind: ConfigMap
    metadata:
      labels:
        app: RELEASE-NAME-jenkins
        app.kubernetes.io/managed-by: Helm
        helm.sh/chart: jenkins-0.7.0
        name: RELEASE-NAME-jenkins
        release-name: RELEASE-NAME
      name: openshift-jenkins-login-plugin-config
